aten/src/ATen/native/mkldnn/Gelu.cpp

```diff
@@ -0,0 +1,89 @@
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/Config.h>
+
+
+#if !AT_MKLDNN_ENABLED()
+
+namespace at { namespace native {
+
+Tensor mkldnn_gelu(const Tensor& input) {
+  TORCH_CHECK(false, "mkldnn_gelu: ATen not compiled with MKLDNN support");
+}
+
+Tensor mkldnn_gelu_backward(const Tensor& grad_output, const Tensor& input) {
+  TORCH_CHECK(false, "mkldnn_gelu_backward: ATen not compiled with MKLDNN support");
+}
+
+}}
+
+#else // AT_MKLDNN_EBABLED
+
+#include <ATen/native/mkldnn/MKLDNNCommon.h>
+#include <ATen/native/mkldnn/Utils.h>
+#include <ATen/native/mkldnn/Gelu.h>
+
+namespace at { namespace native {
+
+Tensor _mkldnn_gelu(const Tensor& input, const Tensor& result) {
+  if (input.scalar_type() == ScalarType::BFloat16) {
+    TORCH_CHECK(mkldnn_bf16_device_check(),
+        "mkldnn_gelu: bf16 path needs the cpu support avx512bw, avx512vl and avx512dq");
```

Comment:
This seems like it ought to be an internal assert, as you should never call this function if `mkldnn_bf16_device_check` fails

Reply:
1. Here is not an internal assert. It will check user behavior to guarantee the device supports avx512.
2. All other mkldnn bf16 will use the same method to deal with device check, because MkldnnCPU path will go directly into mkldnn functions, like mkldnn_gelu().
3. Usually, before calling function, it should call device check first. Here's check has been moved to outside function mkldnn_gelu(). When in CPU path, device check will be done in use_mkldnn(). If check passes and use mkldnn, it will go into _mkldnn_gelu(). When in MkldnnCPU path, device check will be done in mkldnn_gelu(), and then go into _mkldnn_gelu().